digAddressExtractor/address_extractor.py

```diff
@@ -0,0 +1,204 @@
+            flag=1
+            break;
+        start=start-1
+    return flag,start
+
+def getNumNext(text,end,dist):
+    start=end
+    flag=0
+    count=0
+    while end <len(text)-2 and end-start <=dist and text[start]!='\r' and text[start]!='\n':
+        if text[end].isdigit():
+            count+=1
+        if count==5 and text[end+1].isdigit() and (end+1==len(text)-2 or text[end+1]==" " or text[end+1]=="\n" or text[end+1]=="<"):
+            flag=1
+            break;
+        end=end+1
+    return flag,end+1
+
+def getSpace(text,start):
+    while start >0:
+        if text[start-1]==" ":
+            break;
+        start=start-1
+    return start
+
+def extractAddress(text,p,type1,addresses):
+    end=-1
+    m=p.search(text.lower())
+    if m!=None:
+        end=m.span()[0]+len(type1)+1
+    if end !=-1:
+        flag=1
+        flag,bkStart=getNum(text,end-(len(type1)+1),50)
+        if flag==0:
+            start=getSpace(text,end-(len(type1)+2))
+        elif flag==1:
+            flag,start=getNum(text,bkStart-1,10)
+            #print (flag, text[start:end])
+            if flag==0:
+                start=bkStart
+        flag,newEnd=getNumNext(text,end,25)
+        if flag:
+            end=newEnd
+        addresses.add(cleanAddress(text[start:end].lower(),3))
+        m=p.search(text.lower(),end)
+        if m!=None:
+            addresses=extractAddress(text[end:],p,type1,addresses)
+        return addresses
+    return addresses
+#extractAddress("<BR />=?=?=  Table Shower available  =?=?=<br><br>?=Let us make you stress free one day at a time=?<br><br>=?= PUENTE Spa=?= <br><br>TEL:  626-338-8809  <br><br>  1832 Puente Ave. Baldwin Park, CA. 91706  <br><br>Clean Shower Included With Session<br><br>we always hiring beautiful ladies<br><br> Open- 9:00 AM to 9:30 PM<br>  </div>")
+
+
+def processFile(input_file_name,keywords,finalAddressData):
+    input_data=""
+    with open(input_file_name) as data_file:
+        input_data=json.load(data_file)
+    extracted_address=[]
+    for each_type in input_data:
+        for each_a in input_data[each_type]:
+            temp={}
+            temp["input"]=each_a
+            addresses=Set()
+            for each_keyword in keywords:
+                p = re.compile(r'\b%s\b' % each_keyword.lower(), re.I)
+                m=p.search(each_a.lower())
+                if m!=None:
+                    addresses=extractAddress(each_a,p,each_keyword,addresses)
+            temp["address"]=list(addresses)
+            finalAddressData.append(temp)
+
+"""
+Input: Text String
+Output: Json object containing input text string with list of associated present addresses
+
+Uses default keywords embedded in script file
+"""
+def getAddressFromString(text_string):
+    # temp={}
+    #temp["input"]=text_string
+    addresses=Set()
+    for each_keyword in keywords:
+        p = re.compile(r'\b%s\b' % each_keyword.lower(), re.I)
```

Comment:
nice suggestion, but it's not my code, and we need to discuss with the author. I got this address extractor from amandeep at https://github.com/usc-isi-i2/dig-memex-2016/blob/master/code/landmark-extraction/address_extractor.py